src/cli.ts

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/env node
+
+/**
+ * srvx CLI - Universal HTTP server command line interface
+ */
+
+import { parseArgs, showHelp, validateOptions } from "./cli/parser.ts";
+import { detectRuntime, getAdapterImport } from "./cli/runtime.ts";
+import { loadUserFile } from "./cli/loader.ts";
+import packageMetadata from "../package.json" with { type: "json" };
+
+// Get package version
+function getVersion(): string {
```

Comment:
For code style, can you please move internal utils below main export also we can move parser logic to same file as it is relavant.